uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/WorkDay.php

```diff
@@ -165,27 +157,55 @@ private static function validateWorkHours($params) {
       );
     }
 
-    $hasTimesAndBreak = $timeFrom && $timeTo && $break;
-    if($isWorkingDay && !$hasTimesAndBreak) {
+    $hasAllRequiredFields = !is_null($timeFrom) && !is_null($timeTo) && !is_null($break) && !is_null($numberOfHours);
+    if($isWorkingDay && !$hasAllRequiredFields) {
       throw new InvalidWorkDayException(
-        'Time From, Time To and Break are required for Working Days'
+        'Time From, Time To, Break and Number of Hours are required for Working Days'
       );
     }
 
     $hasTimes = !is_null($timeFrom) && !is_null($timeTo);
     if($hasTimes && (strtotime($timeFrom) >= strtotime($timeTo))) {
       throw new InvalidWorkDayException(
-          'Time From should be less than Time To'
+        'Time From should be less than Time To'
       );
     }
 
+    if (!is_null($break)) {
+      if (!is_numeric($break)) {
+        throw new InvalidWorkDayException(
+          'Break should be a valid number'
+        );
+      } else if ($break < 0) {
+        throw new InvalidWorkDayException(
+          'Break cannot be less than 0'
+        );
+      }
+    }
+
     $secondsInWorkingHours = strtotime($timeTo) - strtotime($timeFrom);
     $secondsInBreak = $break * 3600;
     if($hasTimes && ($secondsInBreak >= $secondsInWorkingHours)) {
       throw new InvalidWorkDayException(
-          'Break should be less than the number of hours between Time From and Time To'
+        'Break should be less than the number of hours between Time From and Time To'
       );
     }
+
+    if (!is_null($numberOfHours)) {
```

Comment:
All the conditions can be in their own independent If's, no need for if/elseif, its not as if the validations depend on one another, its just a set of conditions that we need to test.
Will also reduce nesting and improve readability.

Reply:
Done.